app.js

```diff
@@ -253,6 +253,31 @@ app.get("/UserHome", function (req, res) {
     }
 });
 
+app.get("/UserHome", async function(req, res) {
+    let fullNames = {};
+    function getFullName(username){
+        User.findOne({
+            username: username
+        }, function(err, found) {
+            if(found) {
+            return [found.fname, found.lname]
+            }
+        });
+    }
+    for (i in username.connections){
```

Comment:
I didn't define i here. Should be fixed. 
I don't currently have the time to setup the app locally, and the issue I mentioned before still exists. If you want me to test run it on my setup locally, it might be a few days, assuming that I will be able to resolve the issue.